File: Bai3/Python/Sender.py
import pika
import json

connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
channel = connection.channel()
msg = json.dumps({
    'id': 11,
    'packet_no': 100,
    'temperature': 60,
    'humidity': 50,
    'tds': 50,
    'pH': 50
})

channel.queue_declare(queue='hello')
channel.basic_publish(exchange='', routing_key='hello', body=msg)
print("Send: " + msg)
connection.close()

File: Bai3/JS/sender.js
var amqp = require('amqplib/callback_api');

amqp.connect('amqp://localhost', function(error0, connection) {});

amqp.connect('amqp://localhost', function(error0, connection) {
  if (error0) {
    throw error0;
  }
  connection.createChannel(function(error1, channel) {
    if (error1) {
        throw error1;
      }
      var queue = 'hello';
      var msg = {
        "id": 11,
        "packet_no": 100,
        "temperature": 30,
        "humidity": 50,
        "tds": 1100,
        "pH": 50
      }
      channel.assertQueue(queue, {
        durable: false
      });
      var stringData = JSON.stringify(msg);
      console.log('Send data: ' + stringData);
      channel.sendToQueue(queue, Buffer.from(stringData));
  });
  setTimeout(function() {
    connection.close();
    process.exit(0);
}, 500);
});
